src/commands.cc

```diff
@@ -342,6 +342,52 @@ const CommandDesc write_cmd = {
     write_buffer,
 };
 
+void force_write_buffer(const ParametersParser& parser, Context& context, const ShellContext&)
```

Comment:
make `write_buffer` a template function with a `<bool force>` parameter, and pass that force flag to the write_buffer_to_file function, add logic in that function to do the mode change.